### DISCUSSION TITLE
更新日志 - 20230421

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#730] [Announcements]
# 应用：
* QuickStart：
  1. 优化首页加载速度
  2. 修复磁盘挂载对话框输入的挂载点无效的问题
  3. 修复非 iStoreOS 固件（Lede，OpenWRT）下出现 API 500 错误 [openwrt #12388](https://github.com/openwrt/openwrt/pull/12388)

* 统一文件共享：
  1. 删除自带的配置样例（对已配置过的无影响）
  2. 界面增加常见问题的提示 #710

# 设备：
* 新增：恒领科技的 rk3568 系列机型，包括 h66k,h68k,h69k ，共用一个固件，[点此查看](https://fw.koolcenter.com/iStoreOS/h6xk/)。 #585
     > 注意：早期版本的 h66k 不支持共用固件，可能出现网口错乱，如有需要请联系厂商或关注厂商的通知
* r66s/r68s/rk3568-roc-pc：为方便维护，物理 WAN 口统一映射到 eth0，保留配置升级以后逻辑WAN口可能会跟旁边的LAN口交换，请自行在后台改正配置（逻辑WAN口改成eth0，LAN口其他）。如果不介意也可以这样用着，只要交换下WAN口和第一个LAN口的网线。恢复出厂设置或者不保留配置升级都能修复配置。
* x86：22.03.4 测试固件已支持 N5105/N6000/N6005 的硬解，增加 tproxy 和 nftset 。（非此固件，需手动更新，[详情](https://github.com/istoreos/istoreos/issues/716)）

**Comment by wentianci** [comment]
直接安装在笔记本硬盘上，可以识别出网卡驱动，如果是用Ventoy启动istoreOS，会显示无网卡驱动，网上下载好些个版本的openwrt，用ventoy启动openwrt都能识别出网卡的。

**Reply by jjm2473** [reply]
用的是efi还是非efi固件？你是怎么制作ventoy的启动镜像的？直接解压吗

**Reply by wentianci** [reply]
efi和非efi这2中固件都尝试过，直接解压是不行的，有好几种方法，ventoy上有说明一种，还有一种比较简单的是用7-zip压缩软件进行解压，因为7zip压缩软件选择gzip压缩格式压缩的文件后缀就是.gz，istoreOS下载的文件后缀也是.gz，所以用这个压缩软件解压出来的img文件可以在ventoy上使用。(ventoy说明文档上有一篇ventoy启动openwrt的说明，按照上面的方法可以启动istoreOS)

**Reply by jjm2473** [reply]
这个固件支持了ventoy启动，而且不需要按官方文档的特殊处理，解压后就能启动，还支持升级和安装到其他设备上 https://we.tl/t-KdTbt2BGXU

**Reply by wentianci** [reply]
这固件还是不行，界面显示 Run /vtoy/vtoy as init process 这里卡了好一会，然后才显示 crng init done  然后就一直卡这里了。

**Reply by wentianci** [reply]
换了一个比较老的TF卡，就可以了，就是拷贝速度不到2mb/s那叫一个慢，而且容量才4G，不够用。其他版本的istoreOS拷贝到这张卡上还是不行。原来用的ventoy启动盘是固态硬盘的，看来改天得试试哪种品牌的固态硬盘可以用。

**Reply by wentianci** [reply]
又重新试了下，同一个固态硬盘，插笔记本USB启动时，卡住了，屏幕显示crng init done，而其他版本的openwrt固件可以正常启动且正确识别网卡。插入笔记本的sata口时，可以正常启动，而且能正确识别网卡驱动。当然，只有你发的这个固件能识别网卡驱动，其他的istoreOS版本不行。

**Reply by jjm2473** [reply]
奇怪，我用r86s和虚拟机都能正常启动

**Reply by wentianci** [reply]
虚拟机一直都没问题。

**Reply by jjm2473** [reply]
我是说虚拟机启动ventoy再加载istoreos镜像

**Reply by jjm2473** [reply]
你试试最新的ventoy

**Reply by wentianci** [reply]
家里用的小主机一般都是esxi + istoreOS
公司里用的istoreOS之前安装在公司服务器的Oracle VM VirtualBox虚拟机上，一直很稳定
后面公司有不少的老旧笔记本，所以就想装在U盘上。

**Reply by wentianci** [reply]
ventoy-1.0.91 用的这个版本的，应该是最新的版本。至于你说的虚拟机加载ventoy。之前有试过。
我系统是win11的，开启了hyper-V，可能是这个原因，ventoy引导会出错。后来就懒得折腾。

**Reply by jjm2473** [reply]
我用的也是1.0.91，在虚拟机和r86s上都能启动我发的固件

**Reply by jjm2473** [reply]
你直接把固件刷到U盘里能启动吗？如果本身都启动不了，那就不是这里讨论的问题了

**Reply by jjm2473** [reply]
卡住的时候，拍照看看卡在哪了

**Reply by wentianci** [reply]
你发的固件现在可以在ventoyU盘上启动，就是还有一些小问题

**Reply by wentianci** [reply]
有些U盘可以正常启动，有些就卡在上面我说的，屏幕一直显示crng init done

**Reply by jjm2473** [reply]
crng init done前面显示什么，拍照看看

**Reply by jjm2473** [reply]
还有，按我说的，直接刷到U盘看能不能启动，如果也不能启动，那就跟ventoy无关了

**Reply by wentianci** [reply]
![IMG_20230426_122252](https://user-images.githubusercontent.com/59164321/234470672-2f19bb6e-cd2c-4e9a-b46d-5cd58f27119b.jpg)
![IMG_20230426_122300](https://user-images.githubusercontent.com/59164321/234470678-22af9825-390c-4211-878e-facdc926aa16.jpg)
![IMG_20230426_122329](https://user-images.githubusercontent.com/59164321/234470681-6f5a911c-cecd-4e9d-88de-a79de1beb3c4.jpg)

**Reply by wentianci** [reply]
固态硬盘当移动硬盘时，就卡在那了

**Reply by wentianci** [reply]
如果把硬盘直接装在笔记本上，则可以正常启动了。

**Reply by wentianci** [reply]
![IMG_20230426_123039](https://user-images.githubusercontent.com/59164321/234470842-335d478f-d1e6-4cae-bd7c-847ddb43698b.jpg)

**Reply by wentianci** [reply]
用rufus-4.0.exe，将istoreOS写入硬盘，还是跟上面的状况一样：当作移动硬盘时，卡住了；将硬盘直接安装在笔记本上则可以正常启动。

**Reply by wentianci** [reply]
![IMG_20230426_130047](https://user-images.githubusercontent.com/59164321/234474596-3fbfd033-5521-499b-8968-5d7f53361614.jpg)
![IMG_20230426_125814](https://user-images.githubusercontent.com/59164321/234474610-a579d98e-752f-4211-8690-0bebbcc0851a.jpg)

**Reply by wentianci** [reply]
总结以上情况就是：写入固态硬盘时， 从USB启动时会卡住(用的绿联的USB 3.0 to SATA转换器 )，不能正常启动(使用ADATA SP900和KingBank KP330两个品牌的硬盘都是这样)；从本地硬盘启动时就可以正常启动；写入U盘时，有些U盘可以正常启动，有些U盘不能正常启动(多次写入后这个不能正常启动istoreOS的U盘读取不了了)。

**Reply by jjm2473** [reply]
那我估计openwrt 21.02.3有同样的问题，可能是内核没有集成你这个usb硬盘转接线的驱动

**Reply by jjm2473** [reply]
跟硬盘应该没关系，而是usb转接线

**Reply by wentianci** [reply]
应该不至于，有其他版本的openwrt可以正常启动，忘记版本号是多少了。等我重新做好ventoy启动盘后启动看下版本号。

**Reply by wentianci** [reply]
你发的固件，更新完成重启后，识别不了网卡驱动。

**Comment by zjuht** [comment]
有两个问题，1，挂载NFS盘， 重启后，不会自动挂载
                      2，istoreOS 装在一个硬盘里面，在余下的硬盘上划分出20G挂载在某个目录，重启后，就变成挂在一个莫名的盘上
我的机器是R86s 装在emmc上

**Reply by jjm2473** [reply]
1. 不要用NFS，除非你能自己解决问题。
2. 在“挂载点”页面或者首页配置以ssid挂载到固定的挂载点

**Reply by jjm2473** [reply]
如果不使用首页的挂载，可以参考这个 https://doc.linkease.com/zh/guide/easepi/common.html#%E6%8C%82%E8%BD%BD%E7%A1%AC%E7%9B%98

**Reply by zjuht** [reply]
> 1. 不要用NFS，除非你能自己解决问题。
> 2. 在“挂载点”页面或者首页配置以ssid挂载到固定的挂载点

资源都在NAS上，只能通过NFS或者共享目录方式

**Reply by zjuht** [reply]
解决了，在/etc/rc.local 里面加入挂载命令，界面上的不能自动挂载。

**Reply by zjuht** [reply]
N5105的什么时候出稳定版，另外现在固件支持自动更新么？

**Reply by zjuht** [reply]
发现还是不行，无论/etc/rc.local， 还是/etc/fstab 加了，好像都还是不行

**Reply by jjm2473** [reply]
不能用samba吗

**Reply by zjuht** [reply]
系统->启动项->本地启动脚本 加mount 命令也不行，

**Reply by zjuht** [reply]
samba 挂载不成功

**Reply by zjuht** [reply]
samba 重启后也不会自动挂载，大佬，你可以帮忙看看么，没有这个功能，感觉当主服务器不合适啊，一重启，要先挂载，再重启container. 家里人根本不会操作

**Comment by jay5jiang** [comment]
R5S无法突破千兆什么时候可以修复？

**Reply by jjm2473** [reply]
你家宽带超过千兆？光猫的lan口也超过千兆？

**Reply by jay5jiang** [reply]
对，联通1000兆宽带，光猫lan口连电脑拨号测速能到1700兆左右，排除过了，就是用r5s拨号500兆左右。。。
